Source/MLX/State.swift

```diff
@@ -46,4 +76,33 @@ extension MLXRandom {
     /// - ``seed(_:)``
     public static let globalState = RandomState()
 
+    /// See ``withRandomState(_:body:)`` and ``resolve(key:)``
+    @TaskLocal
+    static fileprivate var taskLocalRandomState: MLXRandom.RandomState?
+
 }  // MLXRandom
+
+/// Resolve the given key to a concrete MLXArray (random key).
+///
+/// This will use the following values (in order until one is found) to resolve the
+/// random key:
+///
+/// - the passed key, either an ``MLXArray`` or ``MLXRandom/RandomState``
+/// - the task-local ``MLXRandom/RandomState``, see ``withRandomState(_:body:)-6i2p1``
+/// - the global RandomState, ``MLXRandom/globalState``
+public func resolve(key: RandomStateOrKey?) -> MLXArray {
+    key?.asRandomKey() ?? MLXRandom.taskLocalRandomState?.asRandomKey()
+        ?? MLXRandom.globalState.next()
+}
```

Comment:
Why did you make that public? Shouldn't it not be an implementation only function?